include/clangmetatool/meta_tool_factory.h

```diff
@@ -53,6 +58,34 @@ class MetaToolFactory : public clang::tooling::FrontendActionFactory {
 #else
   virtual clang::FrontendAction *create() { return new T(replacements, args); }
 #endif
+
+  /**
+   * Run the tool passed in and export any replacements to a YAML file
+   */
+  int runAndExportChanges(clang::tooling::ClangTool &tool,
+                           llvm::raw_ostream &os) {
```

Comment:
Should this take a filename instead of a stream? Similar to how clang-tidy takes a filename for --export-fixes. Or do you have a particular use case in mind where it is not written to a file?

Reply:
It seemed more generic and useful for testing, though I suppose it's a bit more annoying to use on the tool side.

Also, a very slight consideration from here https://llvm.org/docs/CodingStandards.html#use-raw-ostream
> Public headers should generally not include the raw_ostream header, but use forward declarations and constant references to raw_ostream instances.

Reply:
We could also provide both if you think its helpful, e.g. one that takes a stream and one that takes a filename